### ISSUE TITLE
[Suggestion] Modular configuration using ::class

### ISSUE THREAD

**Issue by gabbydgab** [zendframework/zend-expressive-skeleton#106]
In respect with PR https://github.com/zendframework/zend-expressive-skeleton/pull/54, can we have similar structure with Zend Skeleton App that the modular configuration will be in config/modules.config.php file? In order to leverage the PR [Resolve modules using PHP 5.5 ::class](https://github.com/zendframework/zend-modulemanager/pull/5)

See updated repo branch: https://github.com/gabbydgab/zend-expressive-skeleton/blob/feature/modular/config/modules.config.php

**Features:**
[x] Enabled zendframework/zend-component-installer (as require-dev)
[x] Enabled mtymek/expressive-config-manager (as required)
[x] Load modules in config/modules.config.php file
[x] Merged configuration for the loaded modules in config/config.php file (can be renamed as expressive.config.php similar to application.config.php in ZendSkeletonApplication)
[x] PR https://github.com/zendframework/zend-expressive-skeleton/pull/105 included

**Caveat:** 
- need to clean the merging of configuration file [config/config.php](https://github.com/gabbydgab/zend-expressive-skeleton/blob/feature/modular/config/config.php#L23-L26)
- config/modules.config.php file should be using ::class (related PR https://github.com/zendframework/zend-modulemanager/pull/5)

``` php
<?php
return  [
    Zend\Router\ConfigProvider::class,
    Zend\Validator\ConfigProvider::class,
];
```
- it will change the structure of zend-component-installer in pulling 'config-provider' mapper

PS: Haven't updated the test yet. Sorry T_T

**Comment by gabbydgab** [comment]
In addition, composer configuration can either be src/ or module/ (as preffered)

Current configuration is still under src/
```json
"autoload": {
        "psr-4": {
            "App\\": "src/App/",
            "ExpressiveInstaller\\": "src/ExpressiveInstaller/"
        }
    },
    "autoload-dev": {
        "psr-4": {
            "AppTest\\": "test/AppTest/",
            "ExpressiveInstallerTest\\": "test/ExpressiveInstallerTest/"
        }
    },
```

**Comment by xtreamwayz** [comment]
@gabbydgab Thanx for the suggestion. This feature is included in #54.
